### PR TITLE
Add readme to server

### PR DESCRIPTION
**Pull request by jxstxn1**
- Added a readme to the server folder
- sorted pub dependencies
- Added `runs-locally` to server file to be able to switch between docker and pure dart execution
- Removed unused print

### CODE REVIEW

File: template/README.template.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Server Folder
+
+This folder contains the server code for the project.
+It contains the following files:
+
+- `bin/server.dart` - The main entry point for the server.
+- `bin/middlewares.dart` - Contains all the middlewares for the server. Contains all CSP, CORS, and other security headers.
+- `Dockerfile` - The Dockerfile for the server to build it.
```
Comment (robiness) [inline template/README.template.md:8]: Maybe `The Dockerfile to build a docker image containing the server` is more precise 🤔

File: template/README.template.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+## Running the server
+
+The recommended way to run the server is to use the Dockerfile by using the following command:
```
Comment (robiness) [inline template/README.template.md:12]: Maybe `The recommended way to run the server is to run it within docker with the following command:` 🤔

I think for people not familiar with docker the distinction between the actual server and docker is not clear and we should try to make it more clear 🤔

**Review by robiness — APPROVED** [review]
Just two thought on the wording. Great work 👌

File: template/README.template.md (hunk 2 above)
Comment (jxstxn1) [reply template/README.template.md]: Done

File: template/README.template.md (hunk 1 above)
Comment (jxstxn1) [reply template/README.template.md]: Done

**Pull request merged**